modules/cloudsql-instance/main.tf

```diff
@@ -71,7 +71,8 @@ resource "google_sql_database_instance" "primary" {
       ipv4_enabled       = var.network_config.connectivity.public_ipv4
       private_network    = try(var.network_config.connectivity.psa_config.private_network, null)
       allocated_ip_range = try(var.network_config.connectivity.psa_config.allocated_ip_ranges.primary, null)
-      require_ssl        = var.network_config.require_ssl
+      require_ssl        = try(var.ssl.require_ssl, null)
```

Comment:
see comment in variables on setting default to `{}` and preventing null values for `var.ssl`, `try` can also be removed here